### ISSUE TITLE
Downloading From a Private Repo

### ISSUE THREAD

**Issue by psychemedia** [lheagy/nblibrarian#3]
Would it be possible to add support for downloading from private repositories, for example by adding a Github key `.library-config.yml` and using that as part of the call to Github, or otherwise by prompting the user for credentials?

**Comment by lheagy** [comment]
good call! I haven't played around much with the github api for private repos before, but this seems like it should be quite do-able

**Comment by psychemedia** [comment]
I thought I had a gist on something similar but can't spot it offhand. I've also used `keyring` py package before for popping credentials into keychain.
